homeassistant/components/rest/data.py

```diff
@@ -76,3 +79,22 @@ async def async_update(self, log_errors=True):
             self.last_exception = ex
             self.data = None
             self.headers = None
+
+        if self.headers is not None:
+            # If the http request failed, headers will be None
+            content_type = self.headers.get("content-type")
+
+            if content_type and (
+                content_type.startswith("text/xml")
+                or content_type.startswith("application/xml")
+                or content_type.startswith("application/xhtml+xml")
+            ):
+                try:
+                    value = json.dumps(xmltodict.parse(self.data))
+                    _LOGGER.debug("JSON converted from XML: %s", self.data)
+                    self.data = value
+                except ExpatError:
+                    _LOGGER.warning(
+                        "REST xml result could not be parsed and converted to JSON"
+                    )
+                    _LOGGER.debug("Erroneous XML: %s", self.data)
```

Comment:
Could this be moved to a method (`_convert_to_json_if_xml`) that would also be shared with `switch` platform?